tests/test_sequence.rs

```diff
@@ -0,0 +1,207 @@
+use pyo3::class::PySequenceProtocol;
+use pyo3::exceptions::IndexError;
+use pyo3::exceptions::ValueError;
+use pyo3::prelude::*;
+use pyo3::types::IntoPyDict;
+use pyo3::types::PyAny;
+use pyo3::types::PyList;
+
+#[macro_use]
+mod common;
+
+#[pyclass]
+struct ByteSequence {
+    elements: Vec<u8>,
+}
+
+#[pymethods]
+impl ByteSequence {
+    #[new]
+    fn new(obj: &PyRawObject, elements: Option<&PyList>) -> PyResult<()> {
+        obj.init(if let Some(pylist) = elements {
```

Comment:
The `obj.init` should move further in around the `Self { elements: elems }`, i.e. as `obj.init(Self { elements: elems })`